### DISCUSSION TITLE
Why is Safari unable to download apple passes after deploying webApp?

### DISCUSSION THREAD

**Discussion by mbappai** [alexandercerutti/passkit-generator#133] [Q&A]
Hey Alex! So I noticed that I don't have any issues creating a pass dynamically and having it displayed on my mobile device when testing locally on my machine but the try creating a pass on my deployed web App,  I start getting the error that reads "Safari unable to download file" (after I click the "allow" option of the permission modal for site to show pass ofcourse).

I'm suspecting that it might have to do with the `webServiceUrl` and `authenticationToken` fields in the pass.json, but then from what I understood after reading the docs is that, it assumes that the client (my web App) is going to be communicating with the passes in order to be pushing updates to it, hence the requirement to provide a `webServiceUrl` and `authorizationToken`. However, that isn't the case for me, all my app is interested in is using your package (passkit-generator) to generate passes with the provided information for an event that may or may not have a very short expiry date, so we don't need to ever update those passes.

So could the reason why safari is unable to show a pass after deploying web app be because of `webServiceUrl` and `authorizationToken` ?

**Update**

Any idea what this error here means please?

`No "UIViewServicePermittedViewControllerClasses" array in View Service bundle "com.apple.PassbookUIService"'s Info.plist. Cannot check validity of request for class "PKServiceAddPassesViewController". This will become failure in a future build.`

**Comment by alexandercerutti** [comment]
Hey @mbappai, I'm sorry to not have answered you, but the last was one of my worst weekends!
Hope you have solved.

**Reply by mbappai** [reply]
Oh so sorry to hear that man! Yeah I was able to figure out the issue. The issue turned out to be how I'm accessing the certificate files. I was using an absolute path which worked for my local machine but not on the server. So after switching to relative path it all worked.
